### ISSUE TITLE
Cannot start sidekiq in a Sinatra project that uses bugsnag. [uninitialized constant Sidekiq::Middleware::Server]

### ISSUE THREAD

**Issue by le-doude** [bugsnag/bugsnag-ruby#462]
### Description
I am trying to setup Sidekiq to work with a fairly complicated sinatra App running on thin (Cant share code sorry).
I use `sidekiq -r my_sidekiq.rb -C my_config.yml` to run sidekiq.
`my_sidekiq.rb` requires the whole app because there are configurations I want to use in the jobs and in booting sidekiq.
The app itself already uses bugsnag and requires it in the main app definition.
The `require 'bugsnag'` itself fails.

### Issue
When calling `require 'bugsnag'` I get the following error:
```
uninitialized constant Sidekiq::Middleware::Server
/Users/edouard/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/gems/bugsnag-5.3.1/lib/bugsnag/sidekiq.rb:30:in `block (2 levels) in <top (required)>'
/Users/edouard/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/gems/sidekiq-5.0.4/lib/sidekiq.rb:143:in `server_middleware'
/Users/edouard/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/gems/bugsnag-5.3.1/lib/bugsnag/sidekiq.rb:29:in `block in <top (required)>'
/Users/edouard/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/gems/sidekiq-5.0.4/lib/sidekiq.rb:73:in `configure_server'
/Users/edouard/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/gems/bugsnag-5.3.1/lib/bugsnag/sidekiq.rb:28:in `<top (required)>'
/Users/edouard/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/gems/bugsnag-5.3.1/lib/bugsnag.rb:127:in `block in <top (required)>'
/Users/edouard/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/gems/bugsnag-5.3.1/lib/bugsnag.rb:125:in `each'
/Users/edouard/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/gems/bugsnag-5.3.1/lib/bugsnag.rb:125:in `<top (required)>'
[...]
```
I think the missing constant is loaded only in Rails (not sure). So the autoconfig of bugsnag for Sidekiq is failing.
It'd be nice to check if the constant is defined before running that code. Which is what I did to fix the bug and test my code. Commenting the last 4 lines of  `lib/bugsnag/sidekiq.rb` also work.

Only downside I would love to be able to use bugsnag for sidekiq in my project. Is there a way to make this work as is? Am I missing a dependency?

### Environment

Library versions:
```
  * bugsnag (5.3.1)
  * bugsnag-em (0.1)
  * em-http-request (1.1.5)
  * rails-erd (1.5.2)
  * sidekiq (5.0.4)
  * sidekiq-grouping (1.0.9)
  * sinatra (1.4.8)
  * sinatra-activerecord (2.0.13)
  * sinatra-advanced-routes (0.5.3)
  * sinatra-contrib (1.4.7)
```

- OS / OS version:  Mac OS X high sierra | Linux (docker ruby:2.3.4-alpine)
- debug mode or production?: both

### Example Repo

I will try to create one ASAP.

### Example code snippet

```ruby
require 'bugsnag'
```

**Comment by snmaynard** [comment]
Can you take a look at how https://github.com/bugsnag/bugsnag-ruby/tree/master/example/sidekiq differs to your setup? It seems to work using the example for me.

**Comment by Cawllec** [comment]
Hi @le-doude, you can stop middleware being automatically loaded by Bugsnag with the environment variable `BUGSNAG_DISABLE_AUTOCONFIGURE`, as demonstrated [here](https://docs.bugsnag.com/platforms/ruby/other/configuration-options/#custom-middleware-loading).
While I wouldn't advise you to necessarily set this in your overall environment, you can probably set it in the environment you're running your sidekiq scripts in to avoid the problem.

**Comment by Cawllec** [comment]
Note - The above advice only works on newer versions of Bugsnag, and probably wouldn't help with your issue anyway, so sorry about that!

I believe the issue is that you're using Bugsnag `5.3.1` and Sidekiq `5.0.4`.  As noted [here](https://github.com/mperham/sidekiq/wiki/Middleware#notes), Sidekiq 5 no longer ships with middleware out of the box, meaning the call to `Sidekiq::Middleware::Server` will fail in version `5+`.

In later versions of Bugsnag we no-longer try to insert after the logging middleware, and this issue is not present.  All versions of Bugsnag later than `5.3.1` have this issue fixed, so updating to `5.3.3` should fix your issue with no impact on your Bugsnag setup.

**Comment by Cawllec** [comment]
Hi @le-doude, I'm closing this issue for now, but if this error re-occurs please open a new issue here.
